packages/amplify_api/android/src/main/kotlin/com/amazonaws/amplify/amplify_api/AmplifyApiPlugin.kt

```diff
@@ -69,7 +71,15 @@ class AmplifyApiPlugin : FlutterPlugin, MethodCallHandler {
         }
         else if(methodName == "addPlugin"){
             try {
-                Amplify.addPlugin(AWSApiPlugin())
+                val authProviders = ApiAuthProviders.builder()
```

Comment:
request: Could the changes in this file please be removed from the PR? I see the comment about adding to support IAM mode, but strictly speaking, I do not believe these changes are needed for the purpose of the PR: multiple auth modes. I pulled down the PR, omitted this file, and it seemed to work with an app which I configured to have multiple auth modes with API keys and cognito user pools.

Side note: The change in this file, or something similar might be needed for IAM mode but it needs some more research.